test/Date/toStringAndToUTCStringYearPadding.js

```diff
@@ -0,0 +1,56 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// For dates whose years are less than 1000, Date.toString() and Date.toUTCString() should pad the years
+// to 4 digits.
+// See https://github.com/Microsoft/ChakraCore/pull/4067
+
+runTest(new Date("0001-10-13T05:16:33Z"),
+    "Fri Oct 12 0001 22:16:33 GMT-0700 (Pacific Daylight Time)",
+    "Sat, 13 Oct 0001 05:16:33 GMT");

```

Comment:
I think this test would read better if the definition of `runTest` were at the top of the file.

Also, you might consider using UnitTestFramework.js (see usage in other tests) and using asserts instead of rolling your own messages for failures.